resin/main.go

```diff
@@ -124,6 +133,16 @@ func main() {
 		fixPathCheckExists("auth-failed-banner")
 	}
 
+	shellCreds := &syscall.Credential{}
+	if viper.IsSet("shell-uid") || viper.IsSet("shell-gid") {
```

Comment:
Do you need to check if the value is set? Can you just check if the value is greater than 0 instead as the default value is -1?